pkg/traces/promsdprocessor/prom_sd_processor_test.go

```diff
@@ -196,10 +201,150 @@ func TestOperationType(t *testing.T) {
 				},
 			}
 			p.(*promServiceDiscoProcessor).hostLabels = hostLabels
-			p.(*promServiceDiscoProcessor).processAttributes(attrMap)
+			p.(*promServiceDiscoProcessor).processAttributes(context.TODO(), attrMap)
 
 			actualAttrValue, _ := attrMap.Get(attrKey)
 			assert.Equal(t, tc.expectedValue, actualAttrValue.StringVal())
 		})
 	}
 }
+
+func TestPodAssociation(t *testing.T) {
+	const ipStr = "1.1.1.1"
+
+	testCases := []struct {
+		name            string
+		podAssociations []string
+		ctxFn           func(t *testing.T) context.Context
+		attrMapFn       func(t *testing.T) pdata.AttributeMap
+		expectedIP      string
+	}{
+		{
+			name: "HTTP connection IP",
+			ctxFn: func(t *testing.T) context.Context {
+				r := &http.Request{RemoteAddr: ipStr}
+				c, ok := client.FromHTTP(r)
+				require.True(t, ok)
+				return client.NewContext(context.Background(), c)
+			},
+			attrMapFn:  func(*testing.T) pdata.AttributeMap { return pdata.NewAttributeMap() },
+			expectedIP: ipStr,
+		},
+		{
+			name: "gRPC connection IP",
+			ctxFn: func(t *testing.T) context.Context {
+				grpcCtx := peer.NewContext(context.Background(), &peer.Peer{
+					Addr: &net.TCPAddr{
+						IP:   net.ParseIP(ipStr),
+						Port: 80,
+					},
+				})
+				c, ok := client.FromGRPC(grpcCtx)
+				require.True(t, ok)
+				return client.NewContext(context.Background(), c)
+			},
+			attrMapFn:  func(*testing.T) pdata.AttributeMap { return pdata.NewAttributeMap() },
+			expectedIP: ipStr,
+		},
+		{
+			name:  "ip attribute",
+			ctxFn: func(t *testing.T) context.Context { return context.Background() },
+			attrMapFn: func(*testing.T) pdata.AttributeMap {
+				attrMap := pdata.NewAttributeMap()
+				attrMap.Insert(podAssociationIPLabel, pdata.NewAttributeValueString(ipStr))
```

Comment:
i would change `podAssociationIPLabel` to just `"ip"`. currently the code depends on the configuration strings matching the attributes you are searching for. changing this test to have the static string "ip" will help ensure that this does not drift.

Reply:
Changed, thanks